config/kubesan/kubernetes/csi/node-plugin.yaml

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: Apache-2.0
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: csi-node-plugin
+
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kubesan-csi-node-plugin
+  namespace: system
+  labels:
+    app.kubernetes.io/name: kubesan-csi-node-plugin
+spec:
+  selector:
+    matchLabels:
+     app.kubernetes.io/name: kubesan-csi-node-plugin
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: kubesan-csi-node-plugin
+        app.kubernetes.io/part-of: ksan-storage
+        app.kubernetes.io/component: csi-controller-plugin
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+    spec:
+      serviceAccountName: csi-node-plugin
+      containers:
+        - name: csi-plugin
+          image: kubesan
+          args:
+            - csi-node-plugin
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          volumeMounts:
+            - name: kubelet-dir
+              mountPropagation: Bidirectional # for Filesystem volume mounting
+              mountPath: /var/lib/kubelet
+            - name: socket-dir
+              mountPath: /run/csi
+            - name: dev
+              mountPath: /dev
+            - name: run-kubesan-nbd
+              mountPath: /run/kubesan/nbd
+          securityContext:
+            privileged: true
```

Comment:
I don't know enough about trampoline pods to advise if this is a risk, but I remember that daemonsets are usually the source of privilege trampolines in the cluster. Maybe you linked me to this, but it's probably worth double check that we're using the recommendations explained in https://www.youtube.com/watch?v=qRo1Qw_Hr2A

I think this daemonset also needs cpu architecture boundaries as well.

Reply:
I think we have to set it to privileged in order to access the ksan binaries on the host